File: backend/src/main/java/com/netcracker/backend/service/TaskService.java
package com.netcracker.backend.service;

import com.netcracker.backend.entity.Task;

import java.util.Optional;

public interface TaskService {
    Task save(Task task);

    Iterable<Task> findAll();

    Optional<Task> findById(Long id);

    void deleteById(Long id);
}

File: frontend/src/app/modules/shared/services/task.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Task } from '../models/Task';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  constructor(private http: HttpClient) {

  }

  save(task: Task): Observable<Task> {
    return this.http.post<Task>('/api/task', task);
  }

  findAll(): Observable<Task[]> {
    return this.http.get<Task[]>('/api/task');
  }

  findById(id: number): Observable<Task> {
    return this.http.get<Task>('/api/task/' + id);
  }

  deleteById(id: number): Observable<void> {
    return this.http.delete<void>('/api/task/' + id);
  }
}
